### DISCUSSION TITLE
Direction reversal

### DISCUSSION THREAD

**Discussion by hitwhicks** [theislab/scvelo#730] [Q&A]
Hello!

I am using scVelo with a dataset where I know the terminal clusters ('9','8','6','7'). The tool works great, except it predicts a direction reversal like in a lot of previous cases! I tried running all three models and I get the same result with the velocity directions going from terminal to presumably root clusters. 

![scvelo_vel_stoch](https://user-images.githubusercontent.com/30258030/137606471-f85f40c4-4403-4332-83c1-9048a5c39675.png)

Here are my top 10 velocity genes:
```python
print(adata.var['velocity_genes'].sum(), adata.n_vars)
69 771
```
```python
top_genes = adata.var['fit_likelihood'].sort_values(ascending=False).index[:300]
scv.pl.scatter(adata, basis=top_genes[:10], frameon=False, ncols=5,color='seurat_clusters',save='velocity_genes.png')
```
![scvelo_velocity_genes](https://user-images.githubusercontent.com/30258030/137606557-57a9644d-e8df-44d9-96e1-ba1a5ba94a49.png)

I also looked at the unspliced fractions like in certain other examples like the Erythroid progenitor case: 
```python
counts_s = scv.utils.sum_var(adata.layers['spliced'])
counts_u = scv.utils.sum_var(adata.layers['unspliced'])
fractions_u = counts_u / (counts_s + counts_u)
scv.pl.scatter(adata, color=fractions_u, smooth=True,save='fractions.png')
```

![scvelo_fractions](https://user-images.githubusercontent.com/30258030/137606697-9c568e88-0a5a-4e2b-8781-308246e5cb9d.png)

other than knowing the terminal clusters I will admit not much is known about biology behind the cell type transitions in my case. So I wanted to get your opinion before I went ahead and interpreted it! 

Thanks!

**Comment by hitwhicks** [comment]
> Hello!
> 
> I am using scVelo with a dataset where I know the terminal clusters ('9','8','6','7'). The tool works great, except it predicts a direction reversal like in a lot of previous cases! I tried running all three models and I get the same result with the velocity directions going from terminal to presumably root clusters.
> 
> ![scvelo_vel_stoch](https://user-images.githubusercontent.com/30258030/137606471-f85f40c4-4403-4332-83c1-9048a5c39675.png)
> 
> Here are my top 10 velocity genes:
> 
> ```python
> print(adata.var['velocity_genes'].sum(), adata.n_vars)
> 69 771
> ```
> 
> ```python
> top_genes = adata.var['fit_likelihood'].sort_values(ascending=False).index[:300]
> scv.pl.scatter(adata, basis=top_genes[:10], frameon=False, ncols=5,color='seurat_clusters',save='velocity_genes.png')
> ```
> 
> ![scvelo_velocity_genes](https://user-images.githubusercontent.com/30258030/137606557-57a9644d-e8df-44d9-96e1-ba1a5ba94a49.png)
> 
> I also looked at the unspliced fractions like in certain other examples like the Erythroid progenitor case:
> 
> ```python
> counts_s = scv.utils.sum_var(adata.layers['spliced'])
> counts_u = scv.utils.sum_var(adata.layers['unspliced'])
> fractions_u = counts_u / (counts_s + counts_u)
> scv.pl.scatter(adata, color=fractions_u, smooth=True,save='fractions.png')
> ```
> 
> ![scvelo_fractions](https://user-images.githubusercontent.com/30258030/137606697-9c568e88-0a5a-4e2b-8781-308246e5cb9d.png)
> 
> other than knowing the terminal clusters I will admit not much is known about biology behind the cell type transitions in my case. So I wanted to get your opinion before I went ahead and interpreted it!
> 
> Thanks!

@WeilerP Any input on this? Sorry just very curious! :)

**Reply by WeilerP** [reply]
@hitwhicks, please have a look at _e.g._ #216 or #462. Your phase portraits simply do not have the expected shape s.t. dynamics cannot be inferred (confidently).
